controllers/flinkcluster/flinkcluster_updater.go

```diff
@@ -654,9 +654,15 @@ func (updater *ClusterStatusUpdater) deriveJobStatus() *v1beta1.JobStatus {
 		newJobState = v1beta1.JobStateDeploying
 	// Derive the job state from the observed Flink job, if it exists.
 	case observedFlinkJob != nil:
-		newJobState = getFlinkJobDeploymentState(observedFlinkJob.State)
 		newJob.ID = observedFlinkJob.Id
 		newJob.Name = observedFlinkJob.Name
+		tmpState := getFlinkJobDeploymentState(observedFlinkJob.State)
+		if observedSubmitter.job == nil || tmpState != v1beta1.JobStateSucceeded {
```

Comment:
I think here we need an additional check for application mode. If app mode then we set the state.

Reply:
In application mode, is there still an observedSubmitter?

Reply:
Actually, this should be ok. Yup, there's still an "obeservedSubmitter"